src/sha256.rs

```diff
@@ -66,6 +67,13 @@ impl EngineTrait for HashEngine {
 #[derive(Copy, Clone, PartialEq, Eq, Default, PartialOrd, Ord, Hash)]
 pub struct Hash([u8; 32]);
 
+impl str::FromStr for Hash {
```

Comment:
Can't FromStr be implemented for all `Hash` trait types? Or even for all `FromHex` types?

Reply:
No, because `T: Hash` includes types outside our crate, and `str::FromStr` is not our trait :(.